Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -18,23 +18,18 @@ struct NestingRule: Rule {
 private struct ValidationData {
     private(set) var typeLevel: Int = -1
     private(set) var functionLevel: Int = -1
-    private var declStack = Stack<any DeclSyntaxProtocol>()
+    private(set) var functionOrNotStack = Stack<Bool>()
 
-    var isFunction: Bool {
```

Comment:
I liked this property. Why not keep it to "hide" the implementation details? 😉 

Perhaps name it `lastIsFunction`.